AsyncDisplayKit/ASDisplayNode+Subclasses.h

```diff
@@ -119,6 +119,19 @@ NS_ASSUME_NONNULL_BEGIN
 - (ASLayout *)calculateLayoutThatFits:(ASSizeRange)constrainedSize;
 
 /**
+ * ASDisplayNode's implementation of -layoutThatFits:parentSize: calls this method to resolve the node's size
+ * against parentSize, intersect it with constrainedSize, and call -calculateLayoutThatFits: with the result.
+ *
+ * In certain advanced cases, you may want to customize this logic. Overriding this method allows you to receive all
+ * three parameters and do the computation yourself.
+ *
+ * @warning Overriding this method should be done VERY rarely.
+ */
+- (ASLayout *)calculateLayoutThatFits:(ASSizeRange)constrainedSize
+                     restrictedToSize:(ASSize)size
```

Comment:
Confusing to have a constrained size and restricted size. Let's consider renaming constrainedSize.